src/main/scala/it/pagopa/interop/notifier/service/impl/DynamoIndexService.scala

```diff
@@ -0,0 +1,53 @@
+package it.pagopa.interop.notifier.service.impl
+
+import cats.implicits._
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.notifier.error.NotifierErrors.DynamoReadingError
+import it.pagopa.interop.notifier.model.MessageId.formatMessageId
+import it.pagopa.interop.notifier.model.{IndexQuery, IndexRecord, MessageId}
+import it.pagopa.interop.notifier.service.DynamoService
+import org.scanamo.DynamoReadError.describe
+import org.scanamo._
+import org.scanamo.ops.ScanamoOps
+import org.scanamo.syntax._
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DynamoIndexService(tableName: String)(implicit ec: ExecutionContext)
+    extends DynamoService[IndexRecord, IndexQuery, MessageId] {
+
+  implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  private val dynamoClient: DynamoDbAsyncClient = DynamoDbAsyncClient.create()
+
+  val scanamo: ScanamoAsync        = ScanamoAsync(dynamoClient)
+  val messageIds: Table[MessageId] = Table[MessageId](tableName)
+
+  override def put(message: IndexRecord): Future[Unit] = scanamo.exec(messageIds.put(message.messageId))
+
+  override def get(limit: Int)(query: IndexQuery)(implicit contexts: Seq[(String, String)]): Future[List[MessageId]] = {
```

Comment:
is this actually used?
Is there a case when we receive multiple results filtering by `resourceId`?

Reply:
Probably not, but due to the trait signature this must be overridden.
Are you suggesting to do not implementing it at all?

Reply:
If it's not required, never called and has a logic that does not apply to our model, then yes, I would remove it 😄